### PR TITLE
Replaced Whisper by faster-whisper

### PR DESCRIPTION
**Pull request by EchedelleLR**

### CODE REVIEW

File: voice2text_deltabot/hooks.py [hunk 1]
```diff
@@ -43,8 +43,9 @@ async def log_event(event: AttrDict) -> None:
 async def on_newmsg(event: AttrDict) -> None:
     msg = event.message_snapshot
     if msg.view_type in (const.ViewType.VOICE, const.ViewType.AUDIO):
-        result = MODEL.transcribe(msg.file)
-        await msg.chat.send_message(text=result["text"], quoted_msg=msg.id)
+        segments = MODEL.transcribe(msg.file)
+        result = ''.join([segment.text for segment in segments])
```
Comment (adbenitez) [inline voice2text_deltabot/hooks.py]:
```suggestion
        result = ''.join(segment.text for segment in segments)
```

File: voice2text_deltabot/hooks.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 cli = BotCli("voice2text-bot")
 add_subcommands(cli)
 STATUS = "I am a Delta Chat bot, send me any voice message to convert it to text"
-MODEL: whisper.Whisper = None  # noqa
+MODEL: WhisperModel = None # noqa
```
Comment (adbenitez) [inline voice2text_deltabot/hooks.py]: the comment must have 2 spaces before the #, you should use black to format the code

**Comment by adbenitez** [comment]
what needs to be done: 
* increase minimum python version required to 3.8 in pyproject.toml
* edit the CI workflow to run on python 3.8 instead of 3.7

File: .github/workflows/python-ci.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.7', '3.10']
+        python-version: ['3.8', '3.9', '3.10']
```
Comment (adbenitez) [inline .github/workflows/python-ci.yml]:
```suggestion
        python-version: ['3.8', '3.10']
```

File: README.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ A voice-to-text converter bot for Delta Chat.
 pip install git+https://github.com/deltachat-bot/voice2text_deltabot.git
 ```
 
-The bot uses [Whisper](https://github.com/openai/whisper) to extract the text from voice messages,
+The bot uses [Faster Whisper](https://github.com/guillaumekln/faster-whisper/), which is an improved version of [Whisper](https://github.com/openai/whisper), to extract the text from voice messages,
```
Comment (adbenitez) [inline README.md]:
```suggestion
The bot uses [Faster Whisper](https://github.com/guillaumekln/faster-whisper/) to extract the text from voice messages,
```

File: voice2text_deltabot/const.py [hunk 5]
```diff
@@ -1,2 +1,4 @@
 """Constants"""
 MODEL_CFG_KEY = "ui.voice2text.model"
+DEVICE_CFG_KEY = "ui.voice2text.device"
+COMPUTE_TYPE_CFG_KEY = "ui.voice2text.precision"
```
Comment (adbenitez) [inline voice2text_deltabot/const.py]: why not `PRECISION_TYPE_CFG_KEY` ?

**Comment by adbenitez** [comment]
you should run black, go to the project root and run:

```
pip install -e '.[dev]'
isort .; black .
```

**Comment by EchedelleLR** [comment]
I ran black in local already.

I can do that for all the files if you want.

**Comment by EchedelleLR** [comment]
Use squash and merge for the two commits to be merged as one commit.

**Pull request merged**